### DISCUSSION TITLE
@types/cloudflare has inaccessible types

### DISCUSSION THREAD

**Discussion by dustinlacewell** [DefinitelyTyped/DefinitelyTyped#54592] [Issues with a @types package] [answered]
The `@types/cloudflare` package is structured in a way that prevents importing a majority of the defined types:

```typescript
declare namespace Cloudflare {
  // all the important types defined in here
}

declare class Cloudflare {
  // various client bits here
}

export = Cloudflare
```

I assume it is this way, because the actual `cloudflare` package default exports its client class. However, this severely reduces the value of these typings. I'm not sure what change would be necessary to have the client class continue to be the default export, while all the other types in the namespace become importable.

**Comment by peterblazejewicz** [comment, marked as answer]
@dustinlacewell this seems to be legit construct, it should allow you to use named imports for types:
```ts
import Cloudflare = require('cloudflare');
import {
    RecordTypes,
    ...
} from 'cloudflare';
```

**Reply by dustinlacewell** [reply]
huh, you're right. sorry about this.

**Reply by peterblazejewicz** [reply]
no worries!
